src/provider/rpc.ts

```diff
@@ -100,6 +100,12 @@ export class RpcProvider implements ProviderInterface {
     return this.channel.getBlockWithTxs(blockIdentifier);
   }
 
+  public async getGasPrice(blockIdentifier?: BlockIdentifier) {
+    return this.channel
+      .getBlockWithTxs(blockIdentifier)
```

Comment:
`getBlockWithTxHashes` should generate less amount of data to receive, and will nevertheless return the L1 gas price

Reply:
I checked the response from BlastAPI but I couldn't find details of l1 gas price: 
<img width="811" alt="image" src="https://github.com/starknet-io/starknet.js/assets/64266194/e28ca789-c9c9-42fb-b178-2a5596b7fce3">

Reply:
On my side, with 
```typescript 
const bl=await provider.getBlockWithTxHashes();
```

### https://starknet-sepolia.public.blastapi.io/rpc/v0_7
```
{
  l1_da_mode: 'BLOB',
  l1_data_gas_price: { price_in_fri: '0x91acb74c302d3', price_in_wei: '0x14f44f92900' },
  l1_gas_price: { price_in_fri: '0x2d7f150f9892', price_in_wei: '0x68b5b1857' },
  parent_hash: '0x16cc570bf6841d72a650ac06eefe965f8325c01d80df0ed5de2d3b03d62c6ec',
  sequencer_address: '0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8',
  starknet_version: '0.13.1',
  status: 'PENDING',
  timestamp: 1712319031,
  transactions: [
```

### https://starknet-sepolia.public.blastapi.io/rpc/v0_6
```
{
  l1_gas_price: { price_in_fri: '0x2782d833b2c3', price_in_wei: '0x5b07580b4' },
  parent_hash: '0x2ef3f69a56a90ca2864d4679aa64e857b72b7c9702fb3e3caaf07d863da79d1',
  sequencer_address: '0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8',
  starknet_version: '0.13.1',
  status: 'PENDING',
  timestamp: 1712319274,
  transactions: [
```

So, it should work.

Reply:
Fixed ✅. What caused the problem earlier was because I was using older RPC version. @PhilippeR26